plane/plane-tests/tests/drone_pools.rs

```diff
@@ -0,0 +1,93 @@
+use crate::common::wait_until_backend_terminated;
+use common::test_env::TestEnvironment;
+use plane::types::{
+    ConnectRequest, ExecutorConfig, KeyConfig, PullPolicy, ResourceLimits, SpawnConfig,
+};
+use plane_test_macro::plane_test;
+use serde_json::Map;
+use std::collections::HashMap;
+
+mod common;
+
+#[plane_test]
+async fn drone_pools(env: TestEnvironment) {
+    let controller = env.controller().await;
+    let client = controller.client();
+    let pool = "test".to_string();
+    let _drone = env.drone(&controller).await;
+    let _drone_in_pool = env.drone_in_pool(&controller, &pool).await;
+
+    // Wait for the drone to register. TODO: this seems long.
+    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+
+    tracing::info!("Requesting backend.");
+    let connect_request = ConnectRequest {
+        spawn_config: Some(SpawnConfig {
+            id: None,
+            cluster: Some(env.cluster.clone()),
+            executable: ExecutorConfig {
+                image: "ghcr.io/drifting-in-space/demo-image-drop-four".to_string(),
+                pull_policy: Some(PullPolicy::IfNotPresent),
+                env: HashMap::default(),
+                resource_limits: ResourceLimits::default(),
+                credentials: None,
+            },
+            lifetime_limit_seconds: Some(5),
+            max_idle_seconds: None,
+            use_static_token: false,
+        }),
+        key: Some(KeyConfig {
+            name: "reuse-key".to_string(),
+            namespace: "".to_string(),
+            tag: "".to_string(),
+        }),
+        user: None,
+        auth: Map::default(),
+        pool: None,
+    };
+
+    let response = client.connect(&connect_request).await.unwrap();
+    tracing::info!("Got response.");
+
+    assert!(response.spawned);
+
+    tracing::info!("Requesting backend from pool.");
+    let mut connect_request_with_pool = connect_request.clone();
+    connect_request_with_pool.pool = Some(pool);
+    connect_request_with_pool.key = Some(KeyConfig {
+        name: "pool-specific-key".to_string(),
+        ..connect_request.key.unwrap()
+    });
+
+    let response_from_pool = client.connect(&connect_request_with_pool).await.unwrap();
+    tracing::info!("Got response from pool.");
+
+    assert!(response_from_pool.spawned);
+    let response_from_pool_drone = response_from_pool.drone.clone().unwrap();
+    assert_ne!(response_from_pool_drone, response.drone.unwrap());
```

Comment:
Rather than just testing that these are different, we could affirmatively check that `response.drone` is the ID of `_drone` and `response_from_pool_drone` is the ID of `_drone_in_pool`, however we would need to make the ID of the drone available as a field on `Drone` ([in this file](https://github.com/drifting-in-space/plane/blob/main/plane/src/drone/mod.rs#L142)). I wouldn't be opposed to that, but it could also be a follow-up PR.

Reply:
That's a good callout, I can get working on this in a follow-up PR.

Reply:
Created an issue so this doesn't get lost: https://github.com/drifting-in-space/plane/issues/637

Reply:
This is now fixed in this PR.